File: C/AR_282.c
#include <stdio.h>
#include <stdlib.h>

int voraz(int L, int* v, int n) {
    int ret=0;
    int i  =1;

    int act;
    while (i<n) {
        act=v[i-1];
        while (i<n && v[i]-act<=L) i++;
        ret++;

        i++;
    }

    if (i==n) ret++;

    return ret;
}

int resuelveCaso() {
    int i;

    int N, L;
    if (scanf("%d %d", &N, &L)!=2) return 0;
    
    int* v=(int*)malloc(N*sizeof(int));
    for (i=0;i<N;i++) scanf("%d", &v[i]);

    printf("%d\n", voraz(L, v, N));

    free(v);
    return 1;
}

int main() {
    while (resuelveCaso());

    return 0;
}

File: Java/AR_282.java
import java.util.*;

public class AR_282 {

    static int voraz(int L, List<Integer> v) {
        int ret=0;
        int i=1, n=v.size();
        
        int act;
        while (i<n) {
            act=v.get(i-1);
            while (i<n && v.get(i)-act<=L) i++;
            
            ret++;
            i++;
        }

        if (i==n) ret++;

        return ret;
    }

    static boolean resuelveCaso(Scanner sc) {
        if (!sc.hasNextInt()) return false;
        int N=sc.nextInt();
        int L=sc.nextInt();

        List<Integer> v=new ArrayList<>();
        for (int i=0;i<N;i++) v.add(sc.nextInt());

        System.out.println(voraz(L, v));
        return true;
    }

    // Usando Buffered reader/writer da Memory limit exceeded.
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        while (resuelveCaso(sc));
        
        sc.close();
    }
}